go-controller/pkg/ovn/ovn.go

```diff
@@ -402,7 +402,11 @@ func (oc *DefaultNetworkController) syncNodeGateway(node *corev1.Node, hostSubne
 			return fmt.Errorf("error creating gateway for node %s: %v", node.Name, err)
 		}
 	}
-	return nil
+
+	if util.IsPodNetworkAdvertisedAtNode(oc, node.Name) {
+		return oc.addAdvertisedUDNIsolation(node.Name)
+	}
+	return oc.deleteAdvertisedUDNIsolation(node.Name)
```

Comment:
hmm so we always try to delete? btw is this UDNIsolation also applicable to the default network advertised case so also on the default network switch?